Client/Frontend/Home/TopSites/DataManagement/TopSitesDataUtility.swift

```diff
@@ -0,0 +1,26 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import Storage
+
+protocol TopSitesDataUtilityInterface {
+    func removeSiteMatching(_ engine: OpenSearchEngine?, from sites: [Site]) -> [Site]
+}
+
+struct TopSitesDataUtility: TopSitesDataUtilityInterface {
+    func removeSiteMatching(_ engine: OpenSearchEngine?, from sites: [Site]) -> [Site] {
+        if let defaultSearchEngine = engine {
```

Comment:
pyramids of doom get my eye twitching. lolololol
This could be cleaner with a guard. Someting like `guard let defaultSearchEngines = engine else { return sites }`